packages/store/src/lib/store.spec.ts

```diff
@@ -77,4 +78,36 @@ describe('store', () => {
     expect(store.getValue()).toMatchSnapshot();
     expect(spy).toHaveBeenCalledTimes(3);
   });
+
+  it('elfHooks.preStoreUpdate should be called if defined', () => {
+    elfHooks.registerPreStoreUpdate<typeof state>(() => {
+      return { filter: 'elf' };
+    });
+
+    const { state, config } = createState(
```

Comment:
Add a spec that the subscriber emits only once. I want to see that the freeze doesn't change any reference.

Reply:
I don't understand. What subscriber? What references should be compared in the `expect` statement?
Please provide an example.

Reply:
```ts
// current state
{
 num: 1,
  prop: {
    nested: {}
  }
}

store.pipe(select(state => state.prop)).subscribe(() => {
 // should be called only once
})

store.update(state => {
  return {
    ...state,
    num:  state.num + 1
  }
})
```